compiler/rustc_const_eval/src/interpret/operand.rs

```diff
@@ -748,6 +748,7 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
     pub(crate) fn const_val_to_op(
         &self,
         val_val: mir::ConstValue<'tcx>,
+        c: Option<mir::Const<'tcx>>,
```

Comment:
This is very strange, taking both a ConstValue and a Const. At the very least it needs a doc comment explaining what callers should do here. But I also haven't understood yet why it is even needed.